### PR TITLE
Add Concepts: Supported Hosts

### PR DESCRIPTION
**Pull request by BillyONeal**
This adds a document describing how we think about which host platforms and configurations are expected to work with vcpkg.

I think Android notes should be added but I need help from @data-queue to add that.

### CODE REVIEW

**Comment by BillyONeal** [comment]
> @BillyONeal : Thanks for your contribution! The author(s) have been notified to review your proposed change. @bion

Does this mean that the 'author' tag at the top should be my GitHub username?

File: vcpkg/concepts/supported-hosts.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+  * apt-based platforms: `apt install git curl zip unzip pkgconfig`
+  * rpm-based platforms: `dnf install git curl zip unzip pkgconfig`
+
+## Curated registry support levels
+
+Support is broken into 4 distinct categories. They are:
+
+* Full support, tested  
+  These configurations are those used to test vcpkg's curated registry on a regular basis. Problems in these
+  configurations are rapidly found and fixed by vcpkg's maintainers and community.
```
Comment (vicroms) [inline vcpkg/concepts/supported-hosts.md:36]:
```suggestion
  The configurations used to test vcpkg's curated registry on a regular basis. Problems in these configurations
  are rapidly found and fixed by vcpkg's maintainers and community..
```

File: vcpkg/concepts/supported-hosts.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+* Full support, expected  
+  These configurations fixes are treated with the same level of priority as 'Full support, tested', but aren't used
+  directly to validate all changes to the curated registry. In general, these platforms are expected to be
+  substantially similar to one of the fully tested configurations.
```
Comment (vicroms) [inline vcpkg/concepts/supported-hosts.md:40]:
```suggestion
  Configurations treated with the same priority as 'Full support, tested' but which are not part of the regular tests.
  In general, these configurations are substantially similar to one of the fully tested configurations.
```

File: vcpkg/concepts/supported-hosts.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+* Community support  
+  These configurations are not actively supported by vcpkg's maintainers, but have substantial community investment
+  where people other than the maintainers fix issues. In general, the vcpkg maintainers will merge changes to fix
+  problems for these platforms, but can't make statements that they should always work. The vcpkg maintainers are
+  interested in moving as many platforms out of community support into full support as possible.
```
Comment (vicroms) [inline vcpkg/concepts/supported-hosts.md:45]:
```suggestion
  Configurations that are not actively supported by vcpkg's maintainers but have substantial community investment.
  vcpkg maintainers accept changes to fix problems for these platforms but don't make guarantees that these 
  configurations work. The vcpkg team is interested in moving as many platforms out of community support
  into full support as possible.
```

File: vcpkg/concepts/supported-hosts.md [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+* Unsupported  
+  These are platforms where things are known to not work for some reason. In general, PRs fixing only these platforms
+  aren't merged.
```
Comment (vicroms) [inline vcpkg/concepts/supported-hosts.md:48]:
```suggestion
  Configurations that are known no to work for some reason. In general, PRs fixing only these platforms
  are not accepted.
```

**Review by vicroms — CHANGES_REQUESTED** [review]
Made some suggestions to avoid passive voice

File: vcpkg/concepts/supported-hosts.md [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+To use vcpkg, a few components need to be installed on the system first.
+
+* Windows: Visual Studio 2015 or later.
+* macOS: XCode Console Tools, tar, git, zip, unzip, and curl. Most ports also need pkg-config.
+We recommend installing these dependencies via Homebrew, such as via a terminal command
+`brew install zip unzip curl pkgconfig`.
+* Linux: A C++ compiler to use, tar, git, zip, unzip, and curl. Most ports also need pkg-config.
+We recommend installing these dependencies with your system package manager.
+  * apt-based platforms: `apt install git curl zip unzip pkgconfig`
+  * rpm-based platforms: `dnf install git curl zip unzip pkgconfig`
```
Comment (dg0yt) [inline vcpkg/concepts/supported-hosts.md:28]: From the issues we know that macOS and Linux users don't get very far without `autoconf autoconf-archive automake make libtool`.

Comment (dg0yt) [reply vcpkg/concepts/supported-hosts.md:28]: And maybe `python3`.

File: vcpkg/concepts/supported-hosts.md [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+updating macOS machines is a manual process, and macOS frequently changes things in ways that break vcpkg's testing. 
+As of this writing, we are using:
+
+* macOS 13.5.2
+* XCode Command Line Tools 14.3.1
+
+### Linux
+
+* The latest LTS release of Ubuntu, currently 22.04.
+
```
Comment (data-queue) [inline vcpkg/concepts/supported-hosts.md:71]:
```suggestion
### Android
* Linux 64-bit Android NDK version 25c
```

File: vcpkg/concepts/supported-hosts.md (hunk 5 above)
Comment (vicroms) [inline vcpkg/concepts/supported-hosts.md:28]:
```suggestion
  * rpm-based platforms: `dnf install git curl zip unzip pkgconfig`

For Linux and macOS users we also recommend to install the following packages: `autoconf`, `autoconf-archive`, `automake`, `make` and `libtool`. While not required, these packages are used often by libraries available in vcpkg.
```

**Review by vicroms — APPROVED** [review]

**Pull request merged**